app/src/main/java/ch/epfl/sdp/geometry/UnboundedArea.java

```diff
@@ -1,11 +1,47 @@
 package ch.epfl.sdp.geometry;
 
+import ch.epfl.sdp.map.MapApi;
+import ch.epfl.sdp.utils.RandomGenerator;
+
 /**
  * Represents an unbounded area. Any point in the 2D plane is inside the area.
  */
-public class UnboundedArea implements Area {
+public class UnboundedArea extends Area {
+
+    /**
+     * The constructor for the unbounded area. We only need a center which can be random
+     */
+    public UnboundedArea() {
+        super(new RandomGenerator().randomGeoPoint());
+    }
+
+    @Override
+    public void shrink(double factor) {
+        //we do not need to shrink anything here
+    }
+
+    @Override
+    public void setShrinkTransition() {
+        //there is no shrink transition here
+    }
```

Comment:
You could have created an interface that implements Area and that extends it.
Then UnboundedArea would simply be an Area and not a ShrinkableArea. ;)

Reply:
You're right, but I will let it as it is for now :)